src/app/modules/main/module.nim

```diff
@@ -1231,9 +1234,36 @@ method onAcceptRequestToJoinSuccess*[T](self: Module[T], communityId: string, me
     item.updatePendingRequestLoadingState(memberKey, false)
 
 method onMembershipStatusUpdated*[T](self: Module[T], communityId: string, memberPubkey: string, status: MembershipRequestState) =
-  let item = self.view.model().getItemById(communityId)
-  if item.id != "":
-    item.updateMembershipStatus(memberPubkey, status)
+  let myPublicKey = singletonInstance.userProfile.getPubKey()
+  let communityDto = self.controller.getCommunityById(communityId)
+
+  if myPublicKey == memberPubkey:
+    case status:
+      of MembershipRequestState.Banned:
+        singletonInstance.globalEvents.showCommunityMemberBannedNotification(fmt "You've been banned from {communityDto.name}", "", communityId)
```

Comment:
This is not only a toast message, this is a system notification too if the application is not active.
The proposed solution has an architectural bug, it will work only if I want to display ephemeral notifications, but if an application is not active - I want to show a system notification, which is not done during sending notifications from QML. To fix this, it will require a big refactoring of the whole ephemeral and system notifications and it is out of the scope of this task

Reply:
Ok so probably my comment should be placed in line `1256`. I'm just talking about the **ephemeral notifications**. We probably have the same issue with **Activity Center** notification strings but that's a different issue that must be, as well, analysed at some point.

In the case of `ephemeral / toast notifications`  the way to go right now, and current implementation to follow, is as I described in my previous message. I just posted the ticket as additional information for you to know that at some point we will work on refactoring this process bc it's not the best. However, while we don't work on this ticket (it's a tech dept and not mandatory) the way to implement ephemeral notifications is the described above.

Reply:
I think we will have some room on the next 2 weeks for having a discussion, the `UI team`, and after that we can present and agree on next steps.

Reply:
> Yeah, taht ticket I linked is not really well described. It mostly just tracks the problem, but I agree that in the end, the way to go is probably to send codes to the UI and let QML do the translations

Yes, exactly, something like this it's what I have in mind.

Reply:
Since we can't translate the OS notifications for now, at least not easily, I'd recommend merging this for now.

We can look to refactor it in the next two weeks after discussing and finding a good solution.

Mykhailo brought up an idea to me that maybe using DOtherSide, we could be able to convert the strings to C++ so that it uses the C++ QT methods for translations? That might be a way forward.

Reply:
Current ephemeral notifications work like this:
1. Send a signal from NIM to QML (from what I saw we have different signals for different notifications)
2. QML creates a msg and sends it back to NIM
3. NIM adds this notification to the model and updates the QML

The current flow skips system notifications, so all functionality that uses this flow does not have system notifications

The perfect solution would be to add some wrapper on a NIM side of c++ tr() fn (but not sure that it is possible and will be recognized by `lupdate`) and get rid of steps 1 and 2

If we go with the existing translation system, we should have a common signal for sending notifications to the QML side and on the QML side parse and form messages based on the type of msg and content of the msg (content can be parsed differently by using data as a JSON)
The created msg must be sent to the NIM side and NIM must decide - to show this msg as an ephemeric or system.
I don't think that fixing my current code will not bring any benefits, only adds complexity to the code, which must be removed after we find a better solution

But if you think that this is a blocker, I can spend time and provide throwing signals for ban/kick/unban from NIM to QML and from QML to NIM, but again, with 99% this code will obsolete in the future

Reply:
added sending ephemeral notification from qml side for lines `1256`

Reply:
Thanks for the modifications. I've left a comment, now that's already merged, FYI for the next time! We now have `ToastsManager.qml` to process all the toasts. There are still some toasts triggered directly from `AppMain` but we are trying to clean them up step by step.